sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/SortOrder.scala

```diff
@@ -71,12 +95,34 @@ case class SortPrefix(child: SortOrder) extends UnaryExpression {
 
   val nullValue = child.child.dataType match {
     case BooleanType | DateType | TimestampType | _: IntegralType =>
-      Long.MinValue
+      if ((child.isAscending && child.nullOrdering == NullFirst) ||
+        (!child.isAscending && child.nullOrdering == NullLast)) {
+          Long.MinValue
```

Comment:
without NULL ordering clause, the default behavior is that for ASC, null rows will be placed first as if smallest value and for DESC, null rows will be placed last as if smallest value also. 

Now with NULL ordering clause, NULL FIRST|LAST is independent of ASC or DESC.